packages/mui-material/src/Tooltip/Tooltip.d.ts

```diff
@@ -160,9 +160,9 @@ export interface TooltipProps extends StandardProps<React.HTMLAttributes<HTMLDiv
    */
   sx?: SxProps<Theme>;
   /**
-   * Tooltip title. Zero-length titles string are never displayed.
+   * Tooltip title. Zero-length titles string, undefined, null and false are never displayed.
    */
-  title: NonNullable<React.ReactNode>;
+  title: React.ReactNode;
```

Comment:
The other changes look good.

However, following the reasoning on https://github.com/mui/material-ui/issues/29696#issuecomment-1200152361 I think that this would be a step forward:

```suggestion
  title: NonNullable<React.ReactNode>;
```

It's not because it's not shown that it should be considered a valid value.